pkg/ccl/streamingccl/streamproducer/span_config_event_stream.go

```diff
@@ -0,0 +1,259 @@
+	s.streamGroup.GoCtx(withErrCapture(s.rfc.Run))
+}
+
+// Next implements tree.ValueGenerator interface.
+func (s *spanConfigEventStream) Next(ctx context.Context) (bool, error) {
+	select {
+	case <-ctx.Done():
+		return false, ctx.Err()
+	case err := <-s.errCh:
+		return false, err
+	case s.data = <-s.streamCh:
+		return true, nil
+	}
+}
+
+// Values implements tree.ValueGenerator interface.
+func (s *spanConfigEventStream) Values() (tree.Datums, error) {
+	return s.data, nil
+}
+
+// Close implements tree.ValueGenerator interface.
+func (s *spanConfigEventStream) Close(ctx context.Context) {
+	s.acc.Close(ctx)
+
+	close(s.doneChan)
+	if err := s.streamGroup.Wait(); err != nil {
+		// Note: error in close is normal; we expect to be terminated with context canceled.
+		log.Errorf(ctx, "span config stream terminated with error %v", err)
+	}
+	s.sp.Finish()
+}
+
+func (s *spanConfigEventStream) handleUpdate(ctx context.Context, update rangefeedcache.Update) {
+	select {
+	case <-ctx.Done():
+	case s.updateCh <- update:
+		if update.Type == rangefeedcache.CompleteUpdate {
+			log.VInfof(ctx, 1, "completed initial scan")
+		}
+	}
+}
+
+// TODO(msbutler): this code is duplicate to event_stream. dedupe.
+func (s *spanConfigEventStream) flushEvent(ctx context.Context, event *streampb.StreamEvent) error {
+	data, err := protoutil.Marshal(event)
+	if err != nil {
+		return err
+	}
+
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case s.streamCh <- tree.Datums{tree.NewDBytes(tree.DBytes(data))}:
+		return nil
+	case <-s.doneChan:
+		return nil
+	}
+}
+
+// streamLoop is the main processing loop responsible for reading buffered rangefeed events,
+// accumulating them in a batch, and sending those events to the ValueGenerator.
+func (s *spanConfigEventStream) streamLoop(ctx context.Context) error {
+	pacer := makeCheckpointPacer(s.spec.Config.MinCheckpointFrequency)
+	batch := streampb.StreamEvent_Batch{}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-s.doneChan:
+			return nil
+		case update := <-s.updateCh:
+			for _, ev := range update.Events {
+				spcfgEvent := ev.(*spanconfigkvsubscriber.BufferEvent)
+				if spcfgEvent.Deletion() {
+					fmt.Printf("\nstreamer found a delete at %s\n", spcfgEvent.GetTarget())
+				}
+				target := spcfgEvent.Update.GetTarget().ToProto()
+				_, tenantID, err := keys.DecodeTenantPrefix(target.GetSpan().Key)
+				if err != nil {
+					return err
+				}
+				if !tenantID.Equal(s.spec.Config.SpanConfigForTenant) {
+					continue
+				}
+				streamedSpanCfgEntry := streampb.StreamEvent_StreamedSpanConfigEntry{
+					SpanConfig: roachpb.SpanConfigEntry{
+						Target: target,
+						Config: spcfgEvent.Update.GetConfig(),
+					},
+					Timestamp: spcfgEvent.Timestamp(),
+				}
+
+				batch.SpanConfigs = append(batch.SpanConfigs, streamedSpanCfgEntry)
+			}
+			if pacer.shouldCheckpoint(update.Timestamp, true) {
```

Comment:
I mildly wonder if we need this pacing. Given that the rangefeedcache is already only pushing us updates when the frontier advances.

Reply:
Added a todo for follow up.